### PR TITLE
NEW: Add type, format, transformer info to plugin pages

### PR DESCRIPTION
**Pull request by Oddant1**
Adds dummy lists of types, formats, and transformers to plugin pages. These list entries currently have no link associated with them. This code is also currently producing this on the "Available plugins" page for plugins with types, formats, or transformers:

___
Selection_104
___

There are no list entries there because the pages the entries would be linking to do not yet exist.

---

```
Co-Authored-By: ChrisKeefe <ChrisKeefe@nau.edu>
```

### CODE REVIEW

**Comment by thermokarst** [comment]
Hey @Oddant1, have you chatted with @ChrisKeefe about reviewing this, first? If he has the availability to take a first pass before me, well, that would be great. Thanks!

**Comment by Oddant1** [comment]
@thermokarst I will definitely do that. Also should add him as a co-author

**Comment by Oddant1** [comment]
Unfortunately GitHub doesn't seem to allow editing of commits to add co-authors from inside GitHub, but I think that last commit should add Chris

**Comment by Oddant1** [comment]
Sigh nope, Hang on

**Comment by thermokarst** [comment]
Just put the `Co-Authored-By` tag in the [main PR description](https://github.com/qiime2/docs/pull/435#issue-326422685) - I will drop that into the commit message when it gets squashed and merged.

**Comment by Oddant1** [comment]
@thermokarst, sounds good, although it also looks like I have the wrong email for Chris? Because it added him and then the commit StackOverflow said I needed to make to not blow everything up took him off :laughing:, but when it added him it didn't create a link to his account

File: source/sphinx_extensions/plugin_directory/extension.py [hunk 1]
```diff
@@ -49,12 +49,18 @@ def generate_rst(app):
         transformers_list = []
 
         for from_type, to_type in plugin.transformers:
-            from_type = repr(from_type).split('.')[-1].replace("'>", '')
-            to_type = repr(to_type).split('.')[-1].replace("'>", '')
+            from_type = repr(from_type) \
```
[the remark this answers is not in the document]
Comment (thermokarst) [reply source/sphinx_extensions/plugin_directory/extension.py]: On second thought, you might just want to look at using `get_view_name`, outright.

**Comment by Oddant1** [comment]
@https://github.com/qiime2/docs/pull/435/commits/7d0436e1d95db9dbe218fc188c4517fb70c541b5 Moves the types, formats, and transformers to their own pages, these pages are not finished at this time. Additionally, the docs do not build without warnings right now, it warns that some of the `.rst` files that are being used as templates are not in toctrees anywhere, and I don't know how to fix it yet.

**Comment by Oddant1** [comment]
I've noticed some oddities in the formats sets we've created, only two items in the Exportable Formats set actually say they're formats. One of them appears to be a pipeline which doesn't seem like it should be there (see the last entry in the list).
____
Selection_002

All these entries in the set came from the dictionary of transformers that already existed on the `PluginManager` object. We just indexed into it using the canonical formats. Seen below is the entry in the transformers dictionary corresponding to that `sklearn.pipeline.Pipeline` entry in the exportable formats set.
____
Selection_003

I'm not sure if this is actually a problem, but it doesn't seem right to me.

**Comment by thermokarst** [comment]
Yikes! Not sure how those all sent as individual messages, meant to submit as a comment-review. Sorry!

**Comment by Oddant1** [comment]
https://github.com/qiime2/docs/pull/435/commits/b9a5f90db4c2c7f10ba76e258424292923c598f6 Still gives warnings on the templates not being included in any toctree

**Comment by lizgehret** [comment]
This is a nice idea, I wonder if we can include something like this after the November release when we re-vamp the dev docs. @ebolyen any thoughts?

**Comment by ebolyen** [comment]
Closing in favor of #543

**Pull request closed without merging**